hashintel/hash
# Generate initial property types of an entity type

## 🌟 What is the purpose of this PR?




This PR adds property type generation for entity types using AI, when creating a new entity type.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1204260760205594/1204460859144791/f) _(internal)_

## 🔍 What does this change?




- modifies the entity type editor to initialise properties from the draft entity (if one is provided)

## 🚀 Has this modified a publishable library?




This PR:


## 📜 Does this require a change to the docs?



Not that I'm aware of.

## ⚠️ Known issues



- Currently it's not possible to get the latest version of a property type using its base URL in the Next.js app. This means all versions of all property types have to be fetched in order to obtain this functionality.

## 🐾 Next steps



- [add error handling](https://app.asana.com/0/1204260760205594/1204461641771566/f)

## 🛡 What tests cover this?



Unfortunately no tests cover the functionality of this PR yet

## ❓ How to test this?



1. Checkout the branch
2. Add an OpenAI API Key to `.env.local`
3. Run the local developer environment following the instructions from the README
4. Create a new entity type, and observe that once the title and description has been filled a list of property types is generated

## 📹 Demo




https://user-images.githubusercontent.com/42802102/234000235-32984611-a042-446e-a105-eb65a3c31fda.mov
